File: entry/check.js
#!/usr/bin/env node

import {readFileSync} from 'fs';
import {Parser, Tree} from 'temper-parsers/all-exports.js';
import {argv} from 'process';

const parser = new Parser(String(readFileSync(argv[2])));

for (let i = 3; i < argv.length; i++) {
    let data = null;
    try {
        data = parser.parse(String(readFileSync(argv[i])));
    } catch (e) {
        console.error(e);
        console.log(`fail ${argv[i]}`);
    }

    if (data != null && !(data instanceof Tree)) {
        console.log(`fail ${argv[i]}`);
    }
}

File: entry/check.py
#!/usr/bin/env python3

import logging
import sys

sys.path.append('temper.out/py/std')
sys.path.append('temper.out/py/temper-core')
sys.path.append('temper.out/py/temper-parsers')

logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, format='%(message)s')

from temper_parsers.src.parser import Parser
from temper_parsers.src.common.tree import Node

with open(sys.argv[1]) as f:
    parser = Parser(f.read())

for arg in sys.argv[2:]:
    with open(arg) as f:
        src = f.read()
    try:
        data = parser.parse(src)
    except RuntimeError:
        data = None
    if not isinstance(data, Node):
        print(f'fail {arg}')
